mmv1/third_party/terraform/services/storagetransfer/resource_storage_transfer_job_test.go

```diff
@@ -369,6 +369,30 @@ func TestAccStorageTransferJob_notificationConfig(t *testing.T) {
 	})
 }
 
+func TestAccStorageTransferJob_hdfsSource(t *testing.T) {
+	t.Parallel()
+
+	testDataSinkName := acctest.RandString(t, 10)
+	testTransferJobDescription := acctest.RandString(t, 10)
+	testSourceAgentPoolName := fmt.Sprintf("tf-test-source-agent-pool-%s", acctest.RandString(t, 10))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccStorageTransferJobDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccStorageTransferJob_hdfsSource(envvar.GetTestProjectFromEnv(), testDataSinkName, testTransferJobDescription, testSourceAgentPoolName),
+			},
```

Comment:
Since all fields are optional, It would be good to add update test step?

Reply:
ExactlyOneOf condition exists for transferspec source options

Reply:
hdfs_data_source.path can be updated, Right?

Reply:
updated test case with path and bucket update